### ISSUE TITLE
Moving max-pool layers

### ISSUE THREAD

**Issue by opti-mix** [pytorch/glow#3983]
Max-pooling commutes with a monotonic-increasing activation function such as ReLU. It should be more efficient to apply pooling first (may be it can be implemented as an extension to `sinkDownRescaleToPoolingNode`?). The reason why it is more efficient is simply because the output of pooling has smaller dimensions than its input, which means less work to be done.

Max-pooling could be  eventually moved e.g. before they batch norm as well, but this changes the network and may affect the results, the network may need to be retrained.

See also https://myrtle.ai/how-to-train-your-resnet-8-bag-of-tricks/ for a more detailed description of the idea.

**Comment by ayermolo** [comment]
One caveat is that certain layers can be merged with ReLU, because HW supports it. Now nothing that can't be handled with some extra logic, but just FYI.
